### PR TITLE
wrap aftereach in a context so it runs before the k8s namespace cleanup

### PR DESCRIPTION
**Pull request by bparees**

### CODE REVIEW

**Comment by bparees** [comment]
@gabemontero this change is a bit of a shot in the dark, i think you've had some experience with ginkgo contexts and ordering.. there's a global aftereach that the k8s framework runs which deletes the namespace...i'm trying to ensure our aftereach runs first.  Ginkgo claims AfterEaches run from the inside out, so i'm hoping this will ensure our debug is nested deeper than the k8s aftereach that deletes the namespace.

File: test/extended/builds/build_timing.go [hunk 1]
```diff
@@ -37,66 +37,75 @@ var _ = g.Describe("[Feature:Builds][timing] capture build stages and durations"
...
+				exutil.DumpPodLogsStartingWith("", oc)
+			}
+		})
+
+		g.AfterEach(func() {
+			if g.CurrentGinkgoTestDescription().Failed {
+				exutil.DumpPodStates(oc)
+				exutil.DumpPodLogsStartingWith("", oc)
+			}
+		})
```
Comment (gabemontero) [inline test/extended/builds/build_timing.go]: you've got duplicate AfterEach's here

Comment (bparees) [reply test/extended/builds/build_timing.go]: thanks, will fix.  too much cutting and pasting.

**Comment by bparees** [comment]
> but my recollection was that I had difficulty overriding the behavior of the base k8s plumbing in the extended test framework,

i think the failure i already had on this PR did indicate the namespace was still available (though there was an issue with the pod state dumper itself which i've now fixed).  so i'm at least a little hopeful.

i suppose i should just test it locally by forcing a failure though.

**Comment by bparees** [comment]
yeah confirmed we get a good pod state dump w/ these changes at least in one of the tests after forcing a failure.

**Comment by bparees** [comment]
pushed the fix for the duplicate aftereach

**Comment by gabemontero** [comment]
On Sat, Sep 30, 2017 at 3:24 PM Ben Parees <notifications@github.com> wrote:

> yeah confirmed we get a good pod state dump w/ these changes at least in
> one of the tests after forcing a failure.
>
awesome

**Comment by bparees** [comment]
@gabemontero updated.

**Comment by bparees** [comment]
test code changes only, marking kind/bug.

**Pull request merged**